scipy/constants/tests/test_constants.py

```diff
@@ -1,35 +1,64 @@
                     rtol=0., atol=1e-13)
-    assert_allclose(sc.convert_temperature([491.67, 491.67], 'Rankine', 'C'),
-                    [0., 0.], rtol=0., atol=1e-13)
-    assert_allclose(sc.convert_temperature([491.67, 491.67], 'r', 'F'),
-                    [32., 32.], rtol=0., atol=1e-13)
-    assert_allclose(sc.convert_temperature([32, 32], 'fahrenheit', 'R'),
-                    [491.67, 491.67], rtol=0., atol=1e-13)
-    assert_allclose(sc.convert_temperature([273.15, 273.15], 'K', 'R'),
-                    [491.67, 491.67], rtol=0., atol=1e-13)
-    assert_allclose(sc.convert_temperature([491.67, 0.], 'rankine', 'kelvin'),
-                    [273.15, 0.], rtol=0., atol=1e-13)
+    xp_assert_close(sc.convert_temperature(xp.asarray([491.67, 491.67],
+                                                      dtype=xp.float64),
+                                           'Rankine', 'C'),
+                    xp.asarray([0., 0.], dtype=xp.float64), rtol=0., atol=1e-13)
+    xp_assert_close(sc.convert_temperature(xp.asarray([491.67, 491.67],
+                                                      dtype=xp.float64),
+                                           'r', 'F'),
+                    xp.asarray([32., 32.], dtype=xp.float64), rtol=0., atol=1e-13)
+    xp_assert_close(sc.convert_temperature(xp.asarray([32., 32.], dtype=xp.float64),
+                                           'fahrenheit', 'R'),
+                    xp.asarray([491.67, 491.67], dtype=xp.float64), rtol=0., atol=1e-13)
+    xp_assert_close(sc.convert_temperature(xp.asarray([273.15, 273.15],
+                                                      dtype=xp.float64),
+                                           'K', 'R'),
+                    xp.asarray([491.67, 491.67], dtype=xp.float64), rtol=0., atol=1e-13)
+    xp_assert_close(sc.convert_temperature(xp.asarray([491.67, 0.], dtype=xp.float64),
+                                           'rankine', 'kelvin'),
+                    xp.asarray([273.15, 0.], dtype=xp.float64), rtol=0., atol=1e-13)
+
+
+@array_api_compatible
+def test_convert_temperature_errors(xp):
+    with pytest.raises(NotImplementedError, match="old_scale="):
+        sc.convert_temperature(1, old_scale="cheddar", new_scale="kelvin")
+    with pytest.raises(NotImplementedError, match="new_scale="):
+        sc.convert_temperature(1, old_scale="kelvin", new_scale="brie")
 
 
-def test_lambda_to_nu():
-    assert_equal(sc.lambda2nu([sc.speed_of_light, 1]), [1, sc.speed_of_light])
+@array_api_compatible
+def test_lambda_to_nu(xp):
+    xp_assert_equal(sc.lambda2nu(xp.asarray([sc.speed_of_light, 1])),
+                    xp.asarray([1, sc.speed_of_light]))
 
 
-def test_nu_to_lambda():
-    assert_equal(sc.nu2lambda([sc.speed_of_light, 1]), [1, sc.speed_of_light])
+@array_api_compatible
+def test_nu_to_lambda(xp):
+    xp_assert_equal(sc.nu2lambda(xp.asarray([sc.speed_of_light, 1])),
+                    xp.asarray([1, sc.speed_of_light]))
 
```

Comment:
can we add `np_only` tests for each of these tests, without the `xp.asarray`s? For the temperatures, can just make it one comparison. Of course, the implementation here does still work with the array-likes we want it to, but that relies on `array_namespace` behaving and being used correctly. It would be nice to have the test coverage (more or less) strictly increase.